R/as.treenetdf.R

```diff
@@ -0,0 +1,255 @@
+# convert one of many types to treeNetwork's native' treenetdf form:
+# treenetdf (network) [nodeId|parentId] data.frame
+# leafpathdf (table) [parent|parent|node] data.frame
+# hierarchical nested list (JSON)
+# hclust
+# data.tree Node
+# igraph
+# ape phylo
+
+
+#########################################################################
+# as.treenetdf
+#' @export
+as.treenetdf <- function(data = NULL, ...) {
+  UseMethod("as.treenetdf")
+}
+
+#########################################################################
+# hclust_to_treenetdf
+#' @export
+as.treenetdf.hclust <- function(data, ...) {
+  clustparents <-
+    unlist(sapply(seq_along(data$height), function(i) {
+      parent <- which(i == data$merge)
+      parent <- ifelse(parent > nrow(data$merge), parent - nrow(data$merge), parent)
+      as.integer(ifelse(length(parent) == 0, NA_integer_, parent))
+    }))
+  
+  leaveparents <-
+    unlist(sapply(seq_along(data$labels), function(i) {
+      parent <- which(i * -1 == data$merge)
+      parent <- ifelse(parent > nrow(data$merge), parent - nrow(data$merge), parent)
+      as.integer(ifelse(length(parent) == 0, NA, parent))
+    }))
+  
+  df <- 
+    data.frame(
+      nodeId = 1:(length(data$height) + length(data$labels)),
+      parentId = c(clustparents, leaveparents),
+      name = c(rep('', length(data$height)), data$labels),
+      height = c(data$height, rep(0, length(data$labels)))
+    )
+  
+  if (require('tibble')) { return(tibble::as.tibble(df)) }
```

Comment:
Is this just to return a data frame if the user doesn't have tibble installed? If so, maybe we should import tibble and just use that.